static/src/javascripts/projects/commercial/modules/article-body-adverts.ts

```diff
@@ -114,6 +114,42 @@ const filterNearbyCandidates =
 		);
 	};
 
+/**
+ * Decide whether we have enough space to add additional sizes for a given advert.
+ * This function ensures we don't insert large height ads at the bottom of articles,
+ * when there's not enough room.
+ *
+ * This is a hotfix to prevent adverts at the bottom of articles pushing down content.
+ * Nudge @chrislomaxjones if you're reading this in 2023
+ */
+const decideAdditionalSizes = async (
+	winningPara: HTMLElement,
+	sizes: AdSize[],
+	isLastInline: boolean,
+): Promise<AdSize[]> => {
+	// If this ad isn't the last inline then return all additional sizes
+	if (!isLastInline) {
+		return sizes;
+	}
+
+	// Compute the vertical distance from the TOP of the winning para to the BOTTOM of the article body
+	const distanceFromBottom = await fastdom.measure(() => {
+		const paraTop = winningPara.getBoundingClientRect().top;
```

Comment:
Does this need to take into account scroll height to get the top relative to the page. I'm having purple box flashbacks: https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect

Reply:
I think in this instance we should be okay! Measuring the distance between two coordinates relative to the viewport should give us something that is independent of the viewport itself.